Firely.Fhir.Packages/Core/FileIndex.cs

```diff
@@ -65,16 +65,39 @@ public void Add(PackageReference package, ResourceMetadata metadata)
 
         private static PackageFileReference resolveFromMultipleCandidates(IEnumerable<PackageFileReference> candidates)
         {
-            candidates = candidates.Where(c => c.HasSnapshot == true || c.HasExpansion == true);
-            return candidates.Count() == 1
-                ? candidates.First()
-                : throw new InvalidOperationException("Found multiple conflicting conformance resources with the same canonical url identifier.");
+            //first check which has the file has the highest version
+            List<PackageFileReference> highestVersionedFiles = filterOnHighestVersions(candidates);
+
+            if (highestVersionedFiles.Count == 1)
+            {
+                return highestVersionedFiles[0];
+            }
+
+            //If there are multiple, check if they have a snapshot or expansion, prefer those.
+            else
+            {
+                candidates = filterOnSnapshotOrExapnsions(candidates);
+                return candidates.First();
+            }
         }
 
+        private static List<PackageFileReference> filterOnHighestVersions(IEnumerable<PackageFileReference> candidates) => candidates
+                                                .GroupBy(file => new Version(file.Version is not null ? file.Version : "0.0.0"))
+                                                .OrderByDescending(group => group.Key)
```

Comment:
Will OrderBy here not order lexically?  E.g.  2.0 > 10.0 ?

Reply:
That's why it's OrderByDescending

Reply:
No, but I mean since group.Key (and Version) is a string, the ordering will be based on the alphabet, not on the logical order of a version string (which is more numerical).

Reply:
group.Key is of the Version object, which can be ordered I believe.
I have added more tests and 10.0 > 2.0 in this case as well

Reply:
ah, so Version implements IComparable.  Ok, then it's fine!